pacu/modules/iam__privesc_scan/main.py

```diff
@@ -71,323 +102,640 @@ def main(args, pacu_main: 'Main'):
-        'sts:AssumeRole',
-        'ec2:AssociateIamInstanceProfile',
-        'ec2:DescribeInstances',
-        'ec2:RunInstances',
-        'lambda:CreateEventSourceMapping',
-        'lambda:CreateFunction',
-        'lambda:InvokeFunction',
-        'lambda:UpdateFunctionCode',
-        'lambda:ListFunctions',
-        'dynamodb:CreateTable',
-        'dynamodb:DescribeTables',
-        'dynamodb:ListStreams',
-        'dynamodb:PutItem',
-        'glue:CreateDevEndpoint',
-        'glue:DescribeDevEndpoints',
-        'glue:GetDevEndpoint',
-        'glue:GetDevEndpoints',
-        'glue:UpdateDevEndpoint',
-        'cloudformation:CreateStack',
-        'cloudformation:DescribeStacks',
-        'datapipeline:CreatePipeline',
-        'datapipeline:PutPipelineDefinition',
-        'codestar:CreateProject',
-        'codestar:AssociateTeamMember',
-        'codestar:CreateProjectFromTemplate'
+        "iam:addroletoinstanceprofile",
+        "iam:addusertogroup",
+        "iam:attachgrouppolicy",
+        "iam:attachrolepolicy",
+        "iam:attachuserpolicy",
+        "iam:createaccesskey",
+        "iam:createinstanceprofile",
+        "iam:createloginprofile",
+        "iam:createpolicyversion",
+        "iam:deletepolicyversion",
+        "iam:listattachedgrouppolicies",
+        "iam:listattacheduserpolicies",
+        "iam:listattachedrolepolicies",
+        "iam:listgrouppolicies",
+        "iam:listgroups",
+        "iam:listgroupsforuser",
+        "iam:listinstanceprofiles",
+        "iam:listpolicies",
+        "iam:listpolicyversions",
+        "iam:listrolepolicies",
+        "iam:listroles",
+        "iam:listuserpolicies",
+        "iam:listusers",
+        "iam:passrole",
+        "iam:putgrouppolicy",
+        "iam:putrolepolicy",
+        "iam:putuserpolicy",
+        "iam:setdefaultpolicyversion",
+        "iam:updateassumerolepolicy",
+        "iam:updateloginprofile",
+        "sts:assumerole",
+        "ec2:associateiaminstanceprofile",
+        "ec2:describeinstances",
+        "ec2:runinstances",
+        "lambda:createeventsourcemapping",
+        "lambda:createfunction",
+        "lambda:invokefunction",
+        "lambda:updatefunctioncode",
+        "lambda:listfunctions",
+        "dynamodb:createtable",
+        "dynamodb:describetables",
+        "dynamodb:liststreams",
+        "dynamodb:putitem",
+        "glue:createdevendpoint",
+        "glue:describedevendpoints",
+        "glue:getdevendpoint",
+        "glue:getdevendpoints",
+        "glue:updatedevendpoint",
+        "cloudformation:createstack",
+        "cloudformation:describestacks",
+        "datapipeline:createpipeline",
+        "datapipeline:putpipelinedefinition",
+        "codestar:createproject",
+        "codestar:associateteammember",
+        "codestar:createprojectfromtemplate",
     ]
-    checked_perms = {'Allow': {}, 'Deny': {}}
+
+    checked_perms = {"Allow": {}, "Deny": {}}
+    # user_escalation_methods = {
```

Comment:
I'd delete this commented code.

Reply:
Going to leave this for now since it has some comments about each permission